### PR TITLE
convert api-docs to typescript

### PR DESCRIPTION
**Pull request by nhsiehgit**
converts apidoc scripts to ts

ts-node is necessary to run `.ts` scripts.
We don't want to include the `yarn add ts-node` in case executing the script would install a conflicting version with the package version.

### CODE REVIEW

File: api-docs/index.ts [hunk 1]
```diff
@@ -1,12 +1,11 @@
 /* global process */
 /* eslint import/no-nodejs-modules:0 */
 /* eslint import/no-unresolved:0 */
```
Comment (evanpurkhiser) [inline api-docs/index.ts:3]: Can probably update the comments here to match our other node scripts

File: package.json [hunk 2]
```diff
@@ -225,9 +225,9 @@
     "storybook-build": "build-storybook -c .storybook -o docs-ui/.storybook-out --quiet",
     "webpack-profile": "NO_TS_FORK=1 yarn -s webpack --profile --json > stats.json",
     "install-api-docs": "cd api-docs && yarn install",
-    "build-deprecated-docs": "yarn install-api-docs && node api-docs/index.js api-docs/openapi.json tests/apidocs/openapi-deprecated.json",
-    "diff-docs": "yarn install-api-docs && ts-node api-docs/openapi-diff.ts",
-    "deref-api-docs": "node api-docs/index.js tests/apidocs/openapi-spectacular.json tests/apidocs/openapi-derefed.json",
+    "build-deprecated-docs": "yarn install-api-docs && ./node_modules/.bin/ts-node api-docs/index.ts api-docs/openapi.json tests/apidocs/openapi-deprecated.json",
```
Comment (dashed) [inline package.json:228]: @nhsiehgit It should be fine to simplify `./node_modules/.bin/ts-node` to `ts-node` here

**Comment by nhsiehgit** [comment]
Oh heh
Sorry yea, I was trying to figure out how to get the test builds working
since they didn’t seem happy with ‘ts-node’

I didn’t want to install it globally so tried going for the node_modules
route but it’s not happy with that either :(

Stepped away for now, but will keep poking  on Monday

File: package.json (hunk 2 above)
Comment (billyvg) [reply package.json:228]: I would have `yarn` resolve the location of `ts-node` by doing `yarn ts-node`

**Review by JoshFerge — APPROVED** [review]

File: api-docs/package.json [hunk 3]
```diff
@@ -13,7 +13,8 @@
     "js-yaml": "^3.14.0",
     "json-diff": "^0.7.1",
     "json-refs": "^3.0.15",
-    "openapi-examples-validator": "^4.1.1"
+    "openapi-examples-validator": "^4.1.1",
+    "ts-node": "^10.7.0"
```
Comment (billyvg) [inline api-docs/package.json]: Can we revert this change now since we're running ts-node from root dir?

**Review by billyvg — APPROVED** [review]
lgtm
